src/url_search_params.rs

```diff
@@ -0,0 +1,352 @@
+use crate::ffi;
+
+pub struct URLSearchParams(*mut ffi::ada_url_search_params);
+
+impl Drop for URLSearchParams {
+    fn drop(&mut self) {
+        unsafe { ffi::ada_free_search_params(self.0) }
+    }
+}
+
+impl URLSearchParams {
+    /// Parses an return a URLSearchParams struct.
+    ///
+    /// ```
+    /// use ada_url::URLSearchParams;
+    /// let params = URLSearchParams::parse("a=1&b=2");
+    /// assert_eq!(params.get("a"), Some("1"));
+    /// assert_eq!(params.get("b"), Some("2"));
+    /// ```
+    pub fn parse(input: &str) -> Self {
+        Self(unsafe { ffi::ada_parse_search_params(input.as_ptr().cast(), input.len()) })
+    }
+
+    /// Returns the size of the URLSearchParams struct.
+    ///
+    /// ```
+    /// use ada_url::URLSearchParams;
+    /// let params = URLSearchParams::parse("a=1&b=2");
+    /// assert_eq!(params.size(), 2);
+    /// ```
+    pub fn size(&self) -> usize {
+        unsafe { ffi::ada_search_params_size(self.0) }
+    }
+
+    /// Sorts the keys of the URLSearchParams struct.
+    pub fn sort(&self) {
+        unsafe { ffi::ada_search_params_sort(self.0) }
```

Comment:
Not sure if the underlying `ada_url_search_params` is internally thread-safe?

Usually all mutating methods like `sort`, `append`, etc should take `&mut self` unless you explicitly want to have internal mutability.

Reply:
There is nothing special in Ada related to thread-safety. I've updated all methods to use `&mut self`. In case I might have missed one, please assume I didn't. Thank you!